source/fe/fe_rt_bubbles.cc

```diff
@@ -0,0 +1,288 @@
+                                     FE_RT_Bubbles<dim>::clone() const
+{
+  return std_cxx14::make_unique<FE_RT_Bubbles<dim>>(*this);
+}
+
+
+//---------------------------------------------------------------------------
+// Auxiliary and internal functions
+//---------------------------------------------------------------------------
+
+
+
+template <int dim>
+void
+FE_RT_Bubbles<dim>::initialize_support_points (const unsigned int deg)
+{
+  this->generalized_support_points.resize (this->dofs_per_cell);
+  this->generalized_face_support_points.resize (this->dofs_per_face);
+
+  // Index of the point being entered
+  unsigned int current = 0;
+
+  // On the faces, we choose as many Gauss-Lobatto points
+  // as required to determine the normal component uniquely.
+  // This is the deg of the RT_Bubble element plus one.
+  if (dim>1)
+    {
+      QGaussLobatto<dim-1> face_points (deg+1);
+      Assert (face_points.size() == this->dofs_per_face,
+              ExcInternalError());
+      for (unsigned int k=0; k<this->dofs_per_face; ++k)
+        this->generalized_face_support_points[k] = face_points.point(k);
+      Quadrature<dim> faces = QProjector<dim>::project_to_all_faces(face_points);
+      for (unsigned int k=0;
+           k<this->dofs_per_face*GeometryInfo<dim>::faces_per_cell;
+           ++k)
+        this->generalized_support_points[k] = faces.point(k+QProjector<dim>
+                                                          ::DataSetDescriptor::face(0,
+                                                              true,
+                                                              false,
+                                                              false,
+                                                              this->dofs_per_face));
+
+      current = this->dofs_per_face*GeometryInfo<dim>::faces_per_cell;
+    }
+
+  if (deg==1)
+    return;
+
+  // In the interior, we need anisotropic Gauss-Lobatto quadratures,
+  // one for each direction
+  QGaussLobatto<1> high(deg+1);
+  std::vector<Point<1>> pts = high.get_points();
+  pts.erase(pts.begin());
+  pts.erase(pts.end()-1);
+
+  std::vector<double> wts(pts.size(),1);
+  Quadrature<1> low(pts,wts);
+
+  for (unsigned int d=0; d<dim; ++d)
+    {
+      QAnisotropic<dim> *quadrature;
+      if (dim == 1) quadrature = new QAnisotropic<dim>(high);
+      if (dim == 2) quadrature = new QAnisotropic<dim>(((d==0) ? low : high),
+                                                         ((d==1) ? low : high));
+      if (dim == 3) quadrature = new QAnisotropic<dim>(((d==0) ? low : high),
+                                                         ((d==1) ? low : high),
+                                                         ((d==2) ? low : high));
+      Assert(dim<=3, ExcNotImplemented());
+
+      for (unsigned int k=0; k<quadrature->size(); ++k)
+        this->generalized_support_points[current++] = quadrature->point(k);
+      delete quadrature;
```

Comment:
You can make memory management automatic in this block if you do
```
  std::unique_ptr<QAnisotropic<dim>> quadrature;
  if (dim == 1) quadrature.reset (new ...);
  ...
```
The `unique_ptr` will then automatically delete the variable, and there is no risk of a memory leak if you later changed your mind and rewrote the code in a way where, for example, you need a `break` statement in the loop that surrounds this block.

Reply:
I've done this here https://github.com/dealii/dealii/pull/5726/files for the `fe_raviart_thomas_nodal.cc` file. You can probably just copy-paste my code.